### PR TITLE
Sync develop to master

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description
This is an intermediate branch cut from master which includes fixes to the branch coupling of build and checkout scripts. This coupling can not yet be removed from CI processes because this repository still uses CircleCI. The functionality to decouple CI from branches is currently only available through our GitHub actions

## Motivation and Context
Decouple build/checkout from branch to make release process simpler

## How Has This Been Tested?
CI

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/dockerhub.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Docker Hub build
+
+on:
+  push:
+    branches:
+      - develop
+      - master
+      - "release/*"
+
```
Comment (MishkaMN) [inline .github/workflows/dockerhub.yml:11]: should we add carma-system*?

**Review by MishkaMN — DISMISSED** [review]

**Comment by kjrush** [comment]
```
+ git clone https://github.com/usdot-fhwa-stol/carma-msgs.git /opt/carma/src/CARMAMsgs --branch ''
Cloning into '/opt/carma/src/CARMAMsgs'...
fatal: Remote branch  not found in upstream origin
```

Worth looking into why the branch is being set to the empty string for this build, I'll dig into it a bit too.

File: Dockerfile [hunk 2]
```diff
@@ -12,10 +12,12 @@
...
 #  the License.
 
-FROM usdotfhwastoldev/carma-base:develop as deps
-
+ARG DOCKER_ORG="usdotfhwastoldev"
+ARG DOCKER_TAG="develop"
+FROM ${DOCKER_ORG}/carma-base:${DOCKER_TAG} as deps
 
 FROM deps as setup
+ARG GIT_BRANCH="develop"
```
Comment (kjrush) [inline Dockerfile:20]: I think I figured it out, this sets it as an ARG, but the checkout.bash line below is referencing it as an unitialized environment variable within the run command I think. You might need an `ENV GIT_BRANCH=${GIT_BRANCH}` here to initialize this as an environment variable as well.

**Review by kjrush — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (paulbourelly999) [reply Dockerfile:20]: I just checked the docker build logs and this works fine https://github.com/usdot-fhwa-stol/novatel_gps_driver/actions/runs/8823563427/job/24224278982#step:6:294. Not sure what I am missing here. It correctly checks  out both repos

Comment (paulbourelly999) [reply Dockerfile:20]: Never mind. You were looking at the circleci failure. The issue was not in the dockerfile but in the checkout script. Added a default to develop although I am not sure this CI ever ran successfully.

File: .github/workflows/dockerhub.yml (hunk 1 above)
Comment (paulbourelly999) [reply .github/workflows/dockerhub.yml:11]: Done

File: Dockerfile (hunk 2 above)
Comment (paulbourelly999) [reply Dockerfile:20]: I am considering fixing the CircleCI workflow for sonar scanner analysis out of scope.

**Review by MishkaMN — APPROVED** [review]

**Review by kjrush — APPROVED** [review]

**Pull request merged**